### DISCUSSION TITLE
Disk Space Usage Above Specs

### DISCUSSION THREAD

**Discussion by marcofloriano** [mediacms-io/mediacms#828] [Q&A] [answered]
After uploading a 100 mb video to my Media CMS install (clean, first video, no changes at all), the space consumed by the final files (encoded + HLS + orginal) is actually 4.5 times the original file size (450mb in this example), not 3 times as stated on the docs (should've  300 MB).

Tried first at a Digital Ocean Droplet with Ubuntu 20 and then in a Virtual Box Machine with Ubuntu 20 too.

I need some guidance of yours in order to start searching for the reasons for so large discrepancy

**Comment by mgogoulos** [comment]
The 1/3 ratio (origina_video/total_files) is indicative, so it can be more or less, in my experience it is like an average. 
What are the sizes of the transcoded files in your case, and the original? You can see these numbers if you select the 'Download' button on the media page...

**Reply by marcofloriano** [reply]
![image](https://github.com/mediacms-io/mediacms/assets/205168/cdcc7a55-1e97-48e3-9530-450a503657c0)

Thanks for the help

**Comment by marcofloriano** [comment]
From that alone (download button on front) you can't get the correct space used on the server. You need to check for the spaces in each directory where files are stored after upload and transcode: 

- media_files/encoded 
- media_files/hls
- media_files/original

**Reply by mgogoulos** [reply, marked as answer]
You can still get the correct size, because this will be

1. the original file
2. the total of the encoded files (if you add them)
3. the HLS, which is the same as 2 (the total of the encoded files), but this time they are broken in small chunks

Just sharing some thoughts, 
- The original file cannot be removed now, but it could be an option to remove it after encodings have finished
- The encoded files could also be removed, provided that the HLS are prepared. This will not allow for download of the files (as you can do now, because you cannot download the HLS files and play them locally as mp4 or other video files). Also some browsers might not be able to play HLS (now the browser tries to play the HLS file first, and if it has problems or does not find the file, plays the mp4 video files)
- Not produce HLS files, which is still possible if you remove the Bento4 software support, but this will result in degraded experience in most cases since HLS is better than serving the video files directly (specially on poor internet connections).
just some thoughts
